src/segmented-control/segment.scss

```diff
@@ -32,71 +32,107 @@ $disabled-color: map.get($segment-properties, 'disabled-color');
 $selected-background: map.get($segment-properties, 'active-background');
 $selected-border-color: awsui.$color-border-segment-active;
 $selected-color: awsui.$color-text-segment-active;
+$segment-gap: awsui.$space-static-xxs;
+$segment-divider-width: 1px;
 
 .segment {
   @include styles.font-body-m;
   @include styles.text-flex-wrapping;
   @include styles.font-button;
   letter-spacing: 0.25px;
-  border-block: styles.$control-border-width solid $default-border-color;
-  border-inline: styles.$control-border-width solid $default-border-color;
   padding-block: styles.$control-padding-vertical;
   padding-inline: awsui.$space-button-horizontal;
-  border-inline-end-width: 0;
   background: $default-background;
   color: $default-color;
   overflow: visible;
+
   &:focus {
     outline: none;
   }
-  @for $i from 1 through 6 {
-    &:nth-child(#{$i}) {
-      grid-column: $i;
-      -ms-grid-column: $i;
-    }
+
+  &.disabled {
+    background: $disabled-background;
+    border-color: $disabled-border-color;
+    color: $disabled-color;
   }
+
   @include focus-visible.when-visible {
     @include styles.focus-highlight(awsui.$space-segmented-control-focus-outline-gutter);
   }
-  &:last-child {
-    border-inline-end-width: styles.$control-border-width;
-    border-start-start-radius: 0;
-    border-start-end-radius: awsui.$border-radius-button;
-    border-end-start-radius: 0;
-    border-end-end-radius: awsui.$border-radius-button;
-  }
-  &:first-child {
+
+  &.refresh {
+    border-inline: none;
+    border-block: none;
     border-start-start-radius: awsui.$border-radius-button;
-    border-start-end-radius: 0;
+    border-start-end-radius: awsui.$border-radius-button;
     border-end-start-radius: awsui.$border-radius-button;
-    border-end-end-radius: 0;
+    border-end-end-radius: awsui.$border-radius-button;
+    padding-block: styles.$control-padding-vertical;
+    padding-inline: calc(awsui.$space-button-horizontal - $segment-gap);
+    position: relative;
+    block-size: calc(100% - $segment-gap);
+    display: flex;
+    align-items: center;
+    &:not(:last-child)::after {
+      content: '';
+      position: absolute;
+      inset-inline-end: calc(-1 * ($segment-gap + $segment-divider-width));
+      inset-block-start: 6px;
```

Comment:
is this line actually needed? I can't work out what it's supposed to do, and when toggling it off in dev tools I see no difference

Reply:
Great callout, it's not necessarily needed, as the height of pseudo element is specified properly and its parent has `align-items: center`